### PR TITLE
[+FEATURE] Add checkbox to default contacts form to accept agreement

### PR DESCRIPTION
**Pull request by jwittorf**
Is required by German law due to a ruling by OLG Köln from 11.03.2016
Is disabled by default, needs to be set for German shops

### CODE REVIEW

File: src/app/code/community/FireGento/MageSetup/etc/config.xml [hunk 1]
```diff
@@ -45,6 +45,11 @@
                     <data>FireGento_MageSetup_Helper_Checkout_Data</data>
                 </rewrite>
             </checkout>
+			<contacts>
```
Comment (Schrank) [inline src/app/code/community/FireGento/MageSetup/etc/config.xml]: Please fix formatting

File: src/app/code/community/FireGento/MageSetup/etc/config.xml [hunk 2]
```diff
@@ -14,9 +14,9 @@
  * @category  FireGento
  * @package   FireGento_MageSetup
  * @author    FireGento Team <team@firegento.com>
- * @copyright 2013-2015 FireGento Team (http://www.firegento.com)
+ * @copyright 2013-2016 FireGento Team (http://www.firegento.com)
  * @license   http://opensource.org/licenses/gpl-3.0 GNU General Public License, version 3 (GPLv3)
- * @version   2.2.2
+ * @version   2.2.4
```
Comment (Schrank) [inline src/app/code/community/FireGento/MageSetup/etc/config.xml]: please remove version and @copyright header @therouv , ok with that?

File: src/app/code/community/FireGento/MageSetup/etc/config.xml [hunk 3]
```diff
@@ -352,6 +357,11 @@ Tel: {{var telephone}}
                 <eu_countries>AT,BE,BG,CY,CZ,DE,DK,EE,FI,FR,DE,GR,HU,HR,IE,IT,LV,LT,LU,MT,NL,PL,PT,RO,SK,SI,ES,SE,GB</eu_countries>
             </country>
         </general>
+		<contacts>
```
Comment (Schrank) [inline src/app/code/community/FireGento/MageSetup/etc/config.xml]: same here. Indent is wrong

File: src/app/code/community/FireGento/MageSetup/etc/system.xml [hunk 4]
```diff
@@ -484,5 +484,25 @@
                 </classes>
             </groups>
         </tax>
+		<contacts>
```
Comment (Schrank) [inline src/app/code/community/FireGento/MageSetup/etc/system.xml]: and here :-)

File: src/app/design/frontend/base/default/layout/magesetup.xml [hunk 5]
```diff
@@ -584,4 +584,13 @@
             </action>
         </reference>
     </customer_logged_in>
+
+	<!-- for updated contact form with checkbox to accept agreement -->
```
Comment (Schrank) [inline src/app/design/frontend/base/default/layout/magesetup.xml]: indent wrong

**Comment by Schrank** [comment]
Please make sure, the new files follow https://github.com/firegento/coding-guidelines and https://github.com/firegento/phpcs If you don't want to take care of it, tell me, then I'll do. :)

**Comment by sprankhub** [comment]
Did not try, but looks good :) Thanks!

**Comment by Arn2014** [comment]
Thanks for adding this feature..!
How to add seperate css styling to the added Notice + txt element?
` <strong>
                        <?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('Notice')); ?>:
                    </strong>
                    <?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('You can revoke this agreement at any time and for the future by sending an e-mail to %s.', Mage::getStoreConfig('contacts/email/recipient_email'))); ?>
                </li>`

Thanks!

**Comment by jwittorf** [comment]
You could style the `<strong>` or add a `<span>` around the strings/php echo like this, might want to add some classes if you got other `<strong>` and/or `<span>` elements in your form:

```
<strong class="notice-lead"><?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('Notice')); ?>:</strong>
<span class="notice-text"><?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('You can revoke this agreement at any time and for the future by sending an e-mail to %s.', Mage::getStoreConfig('contacts/email/recipient_email'))); ?></span>
```

Use css like this (might need to adjust to your template):
```
#contactForm strong.notice-lead { color: blue; }
#contactForm span.notice-text {color: green; }
```

**Comment by Arn2014** [comment]
@jwittorf, thanks for providing the code and css examples..works fine!
I still cant get one thing to work in the form.phml - and that is..how to ad \n of <br> to get the whole "Notice section" on a new line so that it displays below the aggreements checkbow and  text..?

Thanks!

**Comment by jwittorf** [comment]
Try this:

```
<strong class="notice-lead"><?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('Notice')); ?>:</strong>
<br>
<span class="notice-text"><?php echo Mage::helper('core')->quoteEscape(Mage::helper('contacts')->__('You can revoke this agreement at any time and for the future by sending an e-mail to %s.', Mage::getStoreConfig('contacts/email/recipient_email'))); ?></span>
```

In html a `<br>` is for new line/break, use `<br />` if you're running xhtml or am I getting you wrong? Could you otherwise provide me with some code from your current template?

**Comment by Arn2014** [comment]
@jwittorf adding "simple" line breaks to the form.phtml did it..everything is alligned properly now!
(..for me as a front-end designer always a bit of confusing when dealing with .phtml files ..)

Thanks again, much appreciated..!

**Comment by kkrieger85** [comment]
Maybe this checkbox is not necessary: 
Please see (in german) https://www.website-check.de/datenschutzrecht/ist-eine-einwilligung-via-checkbox-in-kontaktformularen-notwendig?fb_ref=Default

**Comment by sprankhub** [comment]
Thanks for the information! Law is always debatable, but since this is configurable, I think we do not have to change anything.

**Comment by sprankhub** [comment]
Just checked that again. @jwittorf any reason why you rewrite the `Mage_Contacts_Helper_Data`? I do not see any reason why this is necessary.

**Comment by jwittorf** [comment]
@sprankhub: See line 54 in form.phtml: `<?php if(Mage::helper('contacts')->isAcceptAgreement()): ?>`

I only want to show the new field if it is enabled in the backend and needed an own function. I'm not that deep into backend programming so feel free to correct me if this isn't the propper approach.

**Comment by sprankhub** [comment]
No worries. Rewrites should be prevented whenever possible. A custom helper class would be possible, too. No need for a rewrite here. I just added an issue for that.

**Pull request merged**